docs/Quickstart.md

```diff
@@ -4,180 +4,392 @@
 
 Welcome to the documentation of the Qbeast-Spark project.
 
-In these sections you will find guides to better understand the technology behind an open format and be able to play with it in just a few lines of code. 
+In these sections you will find guides to better understand the technology behind an open format and be able to play with it in just a few lines of code.
 
+Here's a summary of the topics covered in this document:
 
-## Writing
+- [Pre-requisites](#pre-requisites)
+  - [Download JAVA](#download-java)
+  - [Download Apache Spark](#download-apache-spark)
+- [Launch a Spark Shell](#launch-a-spark-shell)
+- [Set up an Application](#set-up-an-application)
+  - [Maven](#maven)
+  - [SBT](#sbt)
+  - [Python](#python)
+  - [Advanced Spark Configuration](#advanced-spark-configuration)
+- [Creating a Table](#creating-a-table)
+  - [Python](#python-1)
+  - [Scala](#scala)
+  - [SQL](#sql)
+  - [Advanced Table Configuration](#advanced-table-configuration)
+- [Append](#append)
+  - [Python](#python-2)
+  - [Scala](#scala-1)
+  - [SQL](#sql-1)
+- [Read](#read)
+  - [Python](#python-3)
+  - [Scala](#scala-2)
+  - [SQL](#sql-2)
+- [QbeastTable API](#qbeasttable-api)
+- [Optimize](#optimize)
+- [Deletes](#deletes)
+- [Cloud Providers Setup](#cloud-providers-setup)
+  - [AWS](#aws)
+  - [GCP](#gcp)
+- [Index Visualizer](#index-visualizer)
+- [Dependencies and Version Compatibility](#dependencies-and-version-compatibility)
 
-Inside the project folder, launch a spark-shell with the required **dependencies**. The following example uses AWS S3 for reading from a **public** dataset.
 
->**ℹ️ Warning: Different cloud providers may require specific versions of Spark or Hadoop, or specific libraries. Refer [here](CloudStorages.md) to check compatibilities.** 
+## Pre-requisites
+
+You can run the qbeast-spark library in two different ways:
+
+1. Interactively: Start the Spark shell (Scala or Python) with Qbeast Spark and Delta Lake and run the code snippets.
+2. As a Project: Set up a Maven or SBT project (Scala or Java), copy the code snippets into a source file, and run the project.
+
+Before starting, ensure you have the following:
+
+- **Java 8+**: Ensure that Java is installed and properly configured.
+- **SBT/Gradle/Maven**: This is for managing dependencies if running in a development environment.
+- **Apache Spark 3.0+**: A Spark installation with support for Scala 2.12
+
+### Download JAVA
+
+As mentioned in the official Apache Spark installation instructions [here](https://spark.apache.org/docs/latest/index.html#downloading), make sure you have a valid Java version installed (8, 11, or 17) and that Java is configured correctly on your system using either the system `PATH` or `JAVA_HOME` environmental variable.
```

Comment:
Do you know if Java 21 or 22 would work too?

Reply:
I think for[ version 4.x of Spark is build for Java 21](https://lists.apache.org/thread/66xx1k9965h440zlrhll9dbcmcbcy8no) , but 3.x still not :(